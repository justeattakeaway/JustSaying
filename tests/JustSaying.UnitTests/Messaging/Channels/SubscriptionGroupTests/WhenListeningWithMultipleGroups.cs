using JustSaying.AwsTools.MessageHandling;
using JustSaying.Messaging.Channels.SubscriptionGroups;
using Newtonsoft.Json;

namespace JustSaying.UnitTests.Messaging.Channels.SubscriptionGroupTests;

public class WhenListeningWithMultipleGroups(ITestOutputHelper testOutputHelper) : BaseSubscriptionGroupTests(testOutputHelper)
{
    private readonly ISqsQueue _queueB = CreateSuccessfulTestQueue("C7506B3F-81DA-4898-82A5-C0293523592A", new TestMessage());
    private readonly ISqsQueue _queueA = CreateSuccessfulTestQueue("EC159934-A30E-45B0-9186-78853F7D3BED", new TestMessage());

    protected override Dictionary<string, SubscriptionGroupConfigBuilder> SetupBusConfig()
    {
        return new Dictionary<string, SubscriptionGroupConfigBuilder>
        {
            {
                "queueA", new SubscriptionGroupConfigBuilder("queueA")
                    .AddQueue(_queueA)
                    .WithPrefetch(5)
                    .WithBufferSize(20)
                    .WithConcurrencyLimit(1)
                    .WithMultiplexerCapacity(30)
            },
            { "queueB", new SubscriptionGroupConfigBuilder("queueB").AddQueue(_queueB) }
        };
    }

    protected override void Given()
    {
        Queues.Add(_queueA);
        Queues.Add(_queueB);
    }

    [Fact]
    public void SubscriptionGroups_OverridesDefaultSettingsCorrectly()
    {
        var interrogationResult = SystemUnderTest.Interrogate();

        var json = JsonConvert.SerializeObject(interrogationResult, Formatting.Indented);

        json.ShouldMatchApproved(c => c.SubFolder("Approvals"));
    }
}