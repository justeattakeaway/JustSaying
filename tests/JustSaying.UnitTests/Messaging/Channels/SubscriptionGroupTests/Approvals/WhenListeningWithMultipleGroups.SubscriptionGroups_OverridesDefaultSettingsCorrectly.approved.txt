{
  "Groups": [
    {
      "Name": "queueA",
      "ConcurrencyLimit": 1,
      "Multiplexer": {
        "ChannelCapacity": 30,
        "ReaderCount": 1
      },
      "ReceiveBuffers": [
        {
          "BufferSize": 20,
          "QueueName": "EC159934-A30E-45B0-9186-78853F7D3BED",
          "Region": "fake-region",
          "Prefetch": 5
        }
      ]
    },
    {
      "Name": "queueB",
      "ConcurrencyLimit": 8,
      "Multiplexer": {
        "ChannelCapacity": 100,
        "ReaderCount": 1
      },
      "ReceiveBuffers": [
        {
          "BufferSize": 10,
          "QueueName": "C7506B3F-81DA-4898-82A5-C0293523592A",
          "Region": "fake-region",
          "Prefetch": 10
        }
      ]
    }
  ]
}